The dependency in the debug build still points to the library without the 'd' suffix:

```
gmake[2]: *** No rule to make target 'lib/gtest-1.8.1/googletest/libgtest.a', needed by 'mixxx-test'.  Stop.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,7 @@ if(POLICY CMP0071)
   cmake_policy(SET CMP0071 NEW)
 endif()
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 
 # Speed up builds on HDDs
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") # GNU is GNU GCC

--- a/build/depends.py
+++ b/build/depends.py
@@ -443,6 +443,10 @@ class Qt(Dependence):
         if build.platform_is_windows:
             # MSVC
             build.env.Append(CXXFLAGS='/std:c++17')
+            # Fix build of googletest 1.10.0
+            # https://developercommunity.visualstudio.com/content/problem/225156/google-test-does-not-work-with-stdc17.html
+            # https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
+            build.env.Append(CXXFLAGS='/Zc:__cplusplus')
         else:
             # GCC/Clang
             build.env.Append(CXXFLAGS='-std=c++17')

--- a/build/depends.py
+++ b/build/depends.py
@@ -439,13 +439,13 @@ class Qt(Dependence):
                 build.env.Append(LINKFLAGS=['-Wl,-rpath,%s' % libdir_path])
                 build.env.Append(LINKFLAGS="-L" + libdir_path)
 
-        # Mixxx requires C++14 support
+        # Mixxx requires C++17 support
         if build.platform_is_windows:
             # MSVC
-            build.env.Append(CXXFLAGS='/std:c++14')
+            build.env.Append(CXXFLAGS='/std:c++17')
         else:
             # GCC/Clang
-            build.env.Append(CXXFLAGS='-std=c++14')
+            build.env.Append(CXXFLAGS='-std=c++17')
 
 
 class TestHeaders(Dependence):

--- a/src/control/controlindicator.cpp
+++ b/src/control/controlindicator.cpp
@@ -35,8 +35,10 @@ void ControlIndicator::slotGuiTick50ms(double cpuTime) {
         case RATIO1TO1_250MS:
             toggle(0.25);
             break;
-        case OFF: // fall through
-        case ON: // fall through
+        case OFF:
+            [[fallthrough]];
+        case ON:
+            [[fallthrough]];
         default:
             // nothing to do
             break;

--- a/src/encoder/encoderbroadcastsettings.cpp
+++ b/src/encoder/encoderbroadcastsettings.cpp
@@ -77,7 +77,8 @@ EncoderSettings::ChannelMode EncoderBroadcastSettings::getChannelMode() const {
     switch(m_pProfile->getChannels()) {
         case 1: return EncoderSettings::ChannelMode::MONO;
         case 2: return EncoderSettings::ChannelMode::STEREO;
-        case 0: // fallthrough
+        case 0:
+        [[fallthrough]];
         default: return EncoderSettings::ChannelMode::AUTOMATIC;
     }
 }

--- a/src/encoder/encodervorbis.cpp
+++ b/src/encoder/encodervorbis.cpp
@@ -63,7 +63,8 @@ void EncoderVorbis::setEncoderSettings(const EncoderSettings& settings)
     switch(settings.getChannelMode()) {
         case EncoderSettings::ChannelMode::MONO:  m_channels = 1; break;
         case EncoderSettings::ChannelMode::STEREO: m_channels = 2; break;
-        case EncoderSettings::ChannelMode::AUTOMATIC: // fallthrough
+        case EncoderSettings::ChannelMode::AUTOMATIC:
+            [[fallthrough]];
         default:
             if (m_bitrate > MONO_BITRATE_THRESHOLD ) {
                 m_channels = 2;

--- a/src/preferences/dialog/dlgprefvinyl.cpp
+++ b/src/preferences/dialog/dlgprefvinyl.cpp
@@ -27,7 +27,6 @@
 #include "vinylcontrol/defs_vinylcontrol.h"
 #include "vinylcontrol/vinylcontrolmanager.h"
 #include "defs_urls.h"
-#include "util/platform.h"
 
 DlgPrefVinyl::DlgPrefVinyl(QWidget * parent, VinylControlManager *pVCMan,
                            UserSettingsPointer  _config)
@@ -349,21 +348,21 @@ void DlgPrefVinyl::VinylTypeSlotApply()
         } else if (ComboBoxVinylSpeed4->currentText() == MIXXX_VINYL_SPEED_45) {
             m_COSpeeds[3]->set(MIXXX_VINYL_SPEED_45_NUM);
         }
-        M_FALLTHROUGH_INTENDED;
+        [[fallthrough]];
     case 3:
         if (ComboBoxVinylSpeed3->currentText() == MIXXX_VINYL_SPEED_33) {
             m_COSpeeds[2]->slotSet(MIXXX_VINYL_SPEED_33_NUM);
         } else if (ComboBoxVinylSpeed3->currentText() == MIXXX_VINYL_SPEED_45) {
             m_COSpeeds[2]->slotSet(MIXXX_VINYL_SPEED_45_NUM);
         }
-        M_FALLTHROUGH_INTENDED;
+        [[fallthrough]];
     case 2:
         if (ComboBoxVinylSpeed2->currentText() == MIXXX_VINYL_SPEED_33) {
             m_COSpeeds[1]->slotSet(MIXXX_VINYL_SPEED_33_NUM);
         } else if (ComboBoxVinylSpeed2->currentText() == MIXXX_VINYL_SPEED_45) {
             m_COSpeeds[1]->slotSet(MIXXX_VINYL_SPEED_45_NUM);
         }
-        M_FALLTHROUGH_INTENDED;
+        [[fallthrough]];
     case 1:
         if (ComboBoxVinylSpeed1->currentText() == MIXXX_VINYL_SPEED_33) {
             m_COSpeeds[0]->slotSet(MIXXX_VINYL_SPEED_33_NUM);

--- a/src/util/platform.h
+++ b/src/util/platform.h
@@ -23,21 +23,4 @@
 #error We do not support your compiler. Please email mixxx-devel@lists.sourceforge.net and tell us about your use case.
 #endif
 
-#if defined(__clang__) && defined(__has_warning)
-#if __has_feature(cxx_attributes) && __has_warning("-Wimplicit-fallthrough")
-#define M_FALLTHROUGH_INTENDED [[clang::fallthrough]]
-#endif
-#elif defined(__GNUC__) && __GNUC__ >= 7
-// Taken from https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough_003d
-// We could also use a comment, but that would require ccache users to set the
-// keep_comments_cpp option. If we switch to C++17, we can use [[fallthough]].
-#define M_FALLTHROUGH_INTENDED __attribute__ ((fallthrough));
-#endif
-
-#ifndef M_FALLTHROUGH_INTENDED
-#define M_FALLTHROUGH_INTENDED \
-  do {                         \
-  } while (0)
-#endif
-
 #endif /* MIXXX_UTIL_PLATFORM_H */

